source/python-sdk/examples/local-agent-evaluations.md

```diff
@@ -0,0 +1,362 @@
+# Evaluating Custom Agents
+
+In addition to being compatible with popular LLM and Agent cloud service providers, the Vijil Python Client enables users to evaluate any agent they have access to with just a few lines of code. The Vijil client can automatically spin up an endpoint for your agent, allowing you to use to test it just like you would any other model. 
+
+## Prerequisites
+
+To get started, make sure you have the following
+
+1. [A Vijil client](run-your-first-test.md). In this topic we'll assume you've instantiated a Vijil client called `client`.
+2. An agent you want to evaluate
+2. If you are not a Vijil Premium user, [an Ngrok Authorization Token](https://dashboard.ngrok.com/get-started/setup/pythonhttps://dashboard.ngrok.com/get-started/setup/python)
+
+````{note}
+Note: We use Ngrok to create private, protected endpoints to your agent. If you're on a Free plan, you'll need to get an Ngrok authorization token. If you're subscribed to the premium version of Vijil, you don't need to worry about this - we take care of it for you.
+````
+
+## Step 1 - Create a Local Agent Executor
+
+In order to make your agent compatible with Vijil's APIs, you need to create an input_adapter and an output_adapter.Like the names imply, the input_adapter transfroms a ChatCompletionRequest from Vijil, into an input that your agent expects, while the output_adapter converts your agent's output into a response that Vijil expects. 
+
+```python
+def example_input_adapter(request: ChatCompletionRequest):
```

Comment:
can we have the `import` statements for `ChatCompletionRequest` and any other vijil objects used here? Just to be maximally clear that these are objects defined by us